zebra-state/src/service.rs

```diff
@@ -252,6 +252,87 @@ impl StateService {
     pub fn utxo(&self, outpoint: &transparent::OutPoint) -> Option<transparent::Output> {
         self.mem.utxo(outpoint).or_else(|| self.sled.utxo(outpoint))
     }
+
+    /// Return an iterator over the relevant chain of the block identified by
+    /// `hash_or_height`.
+    #[allow(dead_code)]
+    pub fn chain(&self, hash_or_height: HashOrHeight) -> Iter<'_> {
+        Iter {
+            service: self,
+            state: State::NonFinalized(hash_or_height),
+        }
+    }
+}
+
+struct Iter<'a> {
+    service: &'a StateService,
+    state: State,
+}
+
+enum State {
```

Comment:
I was confused by the state enum and attribute, I thought they were related to the state service.

```suggestion
struct Iter<'a> {
    service: &'a StateService,
    iter_state: IterState,
}

enum IterState {
```